app/src/main/java/com/hover/stax/bounties/SectionedBounty.java

```diff
@@ -0,0 +1,41 @@
+package com.hover.stax.bounties;
+
+import com.hover.stax.channels.Channel;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+class SectionedBounty {
+	public String header;
+	public List<Bounty> channelBounties;
+
+	public SectionedBounty(String header, List<Bounty> channelBounties) {
+		this.header = header;
+		this.channelBounties = channelBounties;
+	}
+
+	//FIXME: THIS SEARCH SHOULD NOT OCCUR IN THE UI, INSTEAD THE END OBJECT THAT BOUNTY_VIEW_MODEL SENDS.
+	public static List<SectionedBounty> get(List<Channel> channels, List<Bounty> bounties) {
+		List<SectionedBounty> sectionedBounties = new ArrayList<>();
+		Map<Integer, List<Bounty>> channelIdAndBountyListMap = new HashMap<>();
+		Map<Integer, String> channelIdAndTitleMap = new HashMap<>();
+
```

Comment:
Instead of looping twice and creating a new map you should just connect the bounty and channel objects directly